---
fig-cap-location: top
---

# **Day 2 learning objectives**{.unlisted}
We will use the analysis ready gene-count data generated in part 1 of the series to perform differential gene expression and functional enrichment analyses. 

### **Set up R/RStudio**{.unlisted}
- Run RStudio on Nimbus
- Import the read-count matrix in R/RStudio

### **Exploratory analysis**{.unlisted}
- Learn how to perform exploratory analysis using DESeq2 
- Learn how to perform principal component analysis
- Learn how to check the quality of the data and experimental design

### **Differential expression**{.unlisted}
- Create and understand the DESeq2 dataset object
- Identify differentially expressed genes
- Visualise differentially expressed genes 

### **Enrichment anaylsis**{.unlisted}
- Understand and perform functional enrichement analysis using gene ontology terms 
- Visualise enriched gene ontologies and pathways
