---
fig-cap-location: top
from: markdown+emoji
---

## **Day 2 wrap up**

Today, we performed differential expression and functional enrichment analysis using a gene count matrix file generated from nf-core/rnaseq (Day 1). We learnt:

* How to perform exploratory analysis to understand the quality and properties of our data
* How to use DESeq2 to perform differential expression analysis and check that the model was applied correctly
* How to perform gene ontology (GO) enrichment  with clusterProfiler 
* How to visualise our data and results with clusterProfiler:
    + Bar plots
    + Principal components analysis plots
    + Volcano plots
    + Heatmaps 

We performed this analysis with recommended scientific practices that allow for **portable and reproducible analysis**, including:

* Using a singularity container with pre-installed libraries and R environment. 
    + This container is on CVMFS (a filesystem), available through Pawsey Nimbus for your own analysis
+ Using Rmd to run code interactively and include commentary/documentation around the analysis

## **Course survey!** {.unlisted}

Please fill out our **[course survey](https://www.surveymonkey.com/r/rnaseq2022)** before you leave! Help us help you! :grin:
