---
fig-cap-location: top
from: markdown+emoji
---


# **Load the gene-count matrix**

<div class="objectives">  

### **Objectives**
- Load the count matrix in R 
</div>  

<div class="questions">

### **Questions**
- How to import a count-matrix in R?
</div>  

## **The gene-count matrix**

- The gene-count matrix was generated in the below folder using the `nf-core/rnaseq` pipeline:

```default
~/base_directory/working_directory/results/star_salmon/salmon.merged.gene_counts.tsv
```
-  The input to this pipeline were sub-setted fastq files from 6 samples which were mapped to the `chr18` reference of mouse. 
-  This was done so that the excecution of the `nf-core/rnaseq` pipeline can be demonstrated on our small-sized training VMs `(2 cpus and 8 GB RAM)` in quick time `(~ 20 mins)`.
-  **However**, for today's session, we will be using a pre-computed gene-count matrix which was generated by aligning the full set of reads to the complete mouse genome (version mm10). 
-  This file was downloaded previously along with the other files from cloudstor and is named as `Full_count_matrix.txt`. It is available in the path as 

```default
~/base_path/working_directory/Full_count_matrix.txt
```

Please transfer the files `FULL_count_matrix.txt` and `rnaseq_DE_Full_matrix_DryRun.Rmd` to the path `~/base_directory/working_directory/rstudio/` running the following two commands

```default
cp ~/base_path/working_directory/Full_count_matrix.txt/ rstudio/
cp ~/base_path/working_directory/rnaseq_DE_Full_matrix.Rmd rstudio/
```

## **Working in RStudio**

### **R-markdown**

- R Markdown is a file format for making dynamic documents with R.
- An R Markdown document is written in markdown (an easy-to-write plain text format) and contains chunks of embedded R code, like the document shown below.

![RStudio interface](/fig/R_markdown.png)  


### **Open a R-markdown file**

![](/fig/Rmarkdown_open_file.png)  

The Rmarkdown file `rnaseq_DE_Full_matrix.Rmd` contains all R-scipts required for today's analysis. Each individual chunk of code can be run separately by clicking on the arrow next to it top left corner. 
<br>***NOTE** <br>
Please make sure that you run the code chunks in a sequential order. 

### **Load the R-libraries**

- Import all the R libraries/packages which will assist us at various steps in the analysis.

```r
suppressMessages({
library("DESeq2")
library("edgeR")
library("limma")
library("RColorBrewer")
library("gplots")
library("ggplot2")
library("factoextra")
library("devtools")
library("rstudioapi")
library("dplyr")
library("tibble")
library("tidyverse")
library("pheatmap")
library("biomaRt")
library("annotables")
library("org.Mm.eg.db")
library("biobroom")
library("clusterProfiler")
library("pathfindR")
})
```


### **Set the current working directory**

```r
current_path <- getActiveDocumentContext()$path 
setwd(dirname(current_path ))
```
 
### **The count matrix**

- The DESeq2 package expects count data from RNA-seq or another high-throughput sequencing experiment in the form of a matrix of integer values, as input. 
- The value in the i-th row and the j-th column of the matrix tells how many reads can be assigned to gene i in sample j. 
- The values in the matrix should be un-normalized counts or estimated counts of sequencing reads (for single-end RNA-seq) or fragments (for paired-end RNA-seq). 
- The DESeq2 model internally corrects for library size, so transformed or normalized values such as counts scaled by library size should NOT be used as input.

### **Import the count matrix**
- Next we read in the count matrix generated as input the R-script.
- A gene-count matrix was generated using the nf-core/rnaseq pipeline. 
- The input to this pipeline were sub-setted fastq files from 6 samples which were mapped to the chr18 reference of mouse. This was done so that the excecution of the nf-core/rnaseq pipeline can be demonstrated on our small-sized training VMs `(2 cpus and 8 GB RAM)` in quick time `(~ 20 mins)`.
-  **However**, for todayâ€™s session, we will be using a pre-computed gene-count matrix which was generated by aligning the full set of reads to the complete mouse genome.
-  This file was downloaded previously along with the other files from cloudstor and is named as `Full_count_matrix.txt`. It is available in the path as `~/base_path/working_directory/Full_count_matrix.txt`.
-  We need to copy this full gene-count matrix to a specific path which is seeen by the RStudio server. So run the copy command :

```default
cp ~/base_directory/working_directory/FULL_count_matrix.txt ~/base_directory/working_directory/rstudio/
```
 
```r
# Read the Full count matrix (File has been provided for download)
counttable_original<-read.delim("FULL_count_matrix.txt", header=T, row.names=1) 

# View the count matrix
#View(counttable_original)

# Gene symbol as the identifier (when compared to ENSG ID)
counttable<-counttable_original[,c("Symbol","WT1","WT2","WT3","KO1","KO2","KO3")]
row.names(counttable) <- NULL

# Convert Column'GeneSymbol' to rowname)
rownames(counttable) <- counttable$Symbol
counttable<-counttable[,c("WT1","WT2","WT3","KO1","KO2","KO3")]

#View(counttable)
```

![The count matrix](/fig/count_matrix.png)

<div class="keypoints">
### **Key points**
Raw counts can be transformed for doing intial exploratory analysis; however the input to differential expression analysis are the raw counts without transformation.
</div>